estuary/utils/story.py

```diff
@@ -195,9 +205,157 @@ def get_wait_times(self, results):
+            processing_time = completion_time - creation_time
+
+            if processing_time.total_seconds() < 0:
+                id_num = getattr(artifact, artifact.unique_id_property + '_')
+                log.warning(
+                    'A negative processing time was calculated, with a %s with ID %s.',
+                    artifact.__label__, id_num
+                )
+            else:
+                total += processing_time.total_seconds()
+
+        return total
+
+    def get_total_lead_time(self, results):
+        """
+        Get the total lead time - the time from the start of a story until its current state.
+
+        :param list results: contains inflated results from Neo4j
+        :return: the seconds of total time in the story, or None if sufficient data is not available
+        :rtype: int or None
+        """
+        first_artifact = results[0]
+        last_artifact = results[-1]
+        times = {
+            'BugzillaBug': ['creation_time', None],
+            'DistGitCommit': ['commit_date', None],
+            'Advisory': ['created_at', None],
+            'ContainerAdvisory': ['created_at', None],
+            'FreshmakerEvent': ['time_created', 'time_done'],
+            'KojiBuild': ['creation_time', 'completion_time'],
+            'ModuleKojiBuild': ['creation_time', 'completion_time'],
+            'ContainerKojiBuild': ['creation_time', 'completion_time']
+        }
+
+        start_time_key = times[first_artifact.__label__][0]
+        start_time = getattr(first_artifact, start_time_key)
+        if not start_time:
+            id_num = getattr(first_artifact, first_artifact.unique_id_property + '_')
+            log.warning(
+                'While calculating the total lead time, a %s with ID %s was encountered '
+                'without a creation time.',
+                first_artifact.__label__, id_num
+            )
+            return
+        end_time_key = times[last_artifact.__label__][1]
+
+        if end_time_key:
+            end_time = getattr(last_artifact, end_time_key)
+            if not end_time:
+                end_time = datetime.utcnow()
+        elif last_artifact.__label__.endswith('Advisory'):
+            if last_artifact.state in ['SHIPPED_LIVE', 'DROPPED_NO_SHIP']:
+                end_time = getattr(last_artifact, 'status_time')
+            else:
+                end_time = datetime.utcnow()
+        else:
+            end_time = getattr(last_artifact, start_time_key)
+
+        # Remove timezone info so that both are offset naive and thus able to be subtracted
+        start_time = start_time.replace(tzinfo=None)
+        end_time = end_time.replace(tzinfo=None)
+        total = end_time - start_time
+        if total.total_seconds() < 0:
+            id_num = getattr(last_artifact, last_artifact.unique_id_property + '_')
+            log.warning(
+                'A negative total lead time was calculated, in a story ending with a %s with '
```

Comment:
Could you reword the log message so that it mentions the first and last artifact and their IDs? This then gives us enough context to debug.